src/components/CountDown/CountDown.tsx

```diff
@@ -0,0 +1,133 @@
+import React, {useEffect, useRef, useState} from 'react'
+import {AppState, type StyleProp, type ViewStyle, type TextStyle} from 'react-native'
+import styled from 'styled-components/native'
+
+export type CountDownProps = {
+  /*
+  init time countdown by second
+  */
+  initialSeconds: number
+  /*
+  call when count down finish
+  */
+  onFinish?: () => void
+  /*
+  container countdown view style
+  */
+  containerStyle?: StyleProp<ViewStyle>
+  /*
+  text countdown style
+  */
+  textStyle?: StyleProp<TextStyle>
+  /*
+  loop countdown
+  */
+  loop?: boolean
+  /*
+  format countdown
+  */
+  format?: 'MM:SS' | 'HH:MM:SS' | 'DD:HH:MM:SS'
+  /*
+  default count time after milisecond
+  */
+  intervalTimeBySecond?: number
+}
+const CountDown: React.FunctionComponent<CountDownProps> = ({
+  initialSeconds,
+  containerStyle,
+  onFinish,
+  textStyle,
+  loop,
+  format = 'MM:SS',
+  intervalTimeBySecond = 1000,
+}) => {
+  const [seconds, setSeconds] = useState(initialSeconds)
+
+  const appState = useRef(AppState.currentState)
+  const timeEnd = useRef(new Date(new Date().getTime() + initialSeconds * 1000).getTime())
+  const [appStateVisible, setAppStateVisible] = useState(appState.current)
+
+  /*
+    check app state is change
+  */
+  useEffect(() => {
+    const subscription = AppState.addEventListener('change', nextAppState => {
+      appState.current = nextAppState
+      setAppStateVisible(appState.current)
+    })
+    return () => {
+      subscription.remove()
+    }
+  }, [])
+
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      if (seconds >= 0) {
+        const uff = new Date(timeEnd.current - new Date().getTime()).getTime()
+        setSeconds(uff / 1000)
+      }
+      if (seconds < 1) {
+        if (loop) {
+          timeEnd.current = new Date(new Date().getTime() + initialSeconds * 1000).getTime()
+          setSeconds(initialSeconds)
+        } else {
+          clearTimeout(timeout)
+          if (onFinish) {
+            onFinish()
+          }
+        }
+      }
+    }, intervalTimeBySecond)
```

Comment:
the time shouldn't === 1000ms. because `setTimeout` not run after exactly 1000ms. so it will make the time update not correct.